### ISSUE TITLE
Lucid models select all from tables by default?

### ISSUE THREAD

**Issue by SchillingPilot** [adonisjs/core#3476]
I'm putting this in as a feature request as I'm not sure how many people would consider it a 'bug.'

When using the Lucid models, no matter which fields are defined on the models, the standard model static methods (`all`, `findBy`, `find`, etc) do a 

```sql 
SELECT * FROM table_name
```

This issue also appears when using the `.query()` to gain access to the query builder, regardless of what fields are then put in the `select()`. Further aggravating the issue with the query builder being used on the model, it puts all the fields in the `$extras` object on the model.

Is there any way to override this behavior so that only the fields listed in the model are reflected in the resulting query?

Example:
```javascript  
import { BaseModel, column } from '@ioc:Adonis/Lucid/Orm'

export default class Employee extends BaseModel {

    @column({ isPrimary: true })
    public id: number 

    @column()
    public firstName: string

    @column()
    public lastName: string

    @column()
    public position: number
}
```
There may be additional fields, but `const employees = await Employee.all()` should result in the following query (depending on `NamingStrategy`): 
```sql
SELECT id, first_name, last_name, position FROM employees
```
as opposed to
```sql
SELECT * FROM employees
```
The reason this is potentially problematic is because not all columns of a database table may be available to the web application, causing errors when trying to access unauthorized columns. Also, if you work with older databases which might not be normalized you could be overburdening the system with large datasets of irrelevant data.

I searched the documentation looking for a potential solution, but I came up empty. If there isn't one, for any reason, the standard query builder works fine. The models do make for much easier and cleaner coding, however.

Cheers.

**Comment by thetutlage** [comment]
Yes, can be done, but I think it will result in a breaking change for many. So should be published in the next major release.

Also, how about prefixing the table name to the columns to avoid `ambiguous column name` errors in case of joins?
